parsers/AR.py

```diff
@@ -0,0 +1,563 @@
+    
+    find_totals = [i+1 for i,x in enumerate(mapped_data) if x == 'Totales ']
+    total_thermal_generation = sum([mapped_data[i] for i in find_totals])
+    find_nuclear = [i+2 for i, x in enumerate(mapped_data) if x == 'nuclear']
+    total_nuclear_generation = sum([mapped_data[i] for i in find_nuclear])
+    #find_oil = [i+2 for i, x in enumerate(mapped_data) if x == 'oil']
+    #total_oil_generation = sum([mapped_data[i] for i in find_oil])
+
+    #Assume thermal generation is gas unless specifically mapped to another type.
+    total_gas_generation = total_thermal_generation - total_nuclear_generation
+
+    return {'gas': total_gas_generation, 'nuclear': total_nuclear_generation}
+
+
+
+def get_hydro():
+    """Requests hydro generation data then parses, returns a dictionary."""
+
+    s = requests.Session()
+    r = s.get(hurl)
+    pat = re.search("ControlID=[^&]*", r.text).group()
+    spat = re.search("ReportSession=[^&]*", r.text).group()
+    cid = pat.rpartition('=')[2]
+    rs = spat.rpartition('=')[2]
+    full_table = []
+    
+
+    for pagenumber in range(1,2,1):
+        t = s.get(thurl, params = {'ControlID': cid, 'ReportSession': rs, 'PageNumber': '{}'.format(pagenumber)})
+        text_only = webparser(t)
+        if 'En Reserva' in text_only:
+            break
+        full_table.append(text_only)
+
+    data = list(itertools.chain.from_iterable(full_table))
+    formatted_data = dataformat(data)
+    #mapped_data = [power_plant_type.get(x,x) for x in formatted_data]
+    find_hydro = [i+1 for i,x in enumerate(formatted_data) if x == 'Totales ']
+    total_hydro_generation = sum([formatted_data[i] for i in find_hydro])
+
+    return {'hydro': total_hydro_generation}
+
+
+
+def fetch_production(country_code='AR'):
+    """
+    Requests the last known production mix (in MW) of a given country
+    Arguments:
+    country_code (optional) -- used in case a parser is able to fetch multiple countries
+    Return:
+    A dictionary in the form:
+    {
+      'countryCode': 'FR',
+      'datetime': '2017-01-01T00:00:00Z',
+      'production': {
+          'biomass': 0.0,
+          'coal': 0.0,
+          'gas': 0.0,
+          'hydro': 0.0,
+          'nuclear': null,
+          'oil': 0.0,
+          'solar': 0.0,
+          'wind': 0.0,
+          'geothermal': 0.0,
+          'unknown': 0.0
+      },
+      'storage': {
+          'hydro': -10.0,
+      },
+      'source': 'mysource.com'
+    }
+    """
+    gdt = get_datetime()
+    thermal = get_thermal()
+    hydro = get_hydro()
+    production_mix = {
+      'countryCode': country_code,
+      'datetime': gdt['datetime'],
+      'production': {
+          'biomass': None,
+          'coal': None,
+          'gas': thermal['gas'],
+          'hydro': hydro['hydro'],
+          'nuclear': thermal['nuclear'],
+          'oil': None,
+          'solar': None,
+          'wind': None,
+          'geothermal': None,
+          'unknown': None
```

Comment:
Understood.  For types that Argentina has but doesn't provide data for what should I do?